feat: add Pastefy piece

JanHolger opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes #1420

abuaboud commented:
@JanHolger Approved, I added the triggers to piece metadata, can you please confirm that is correct?

JanHolger commented:
> @JanHolger Approved, I added the triggers to piece metadata, can you please confirm that is correct?

Yes, seems like I refactored and forgot to readd the triggers here, thx for fixing it!
